Source/Core/DolphinQt/NetPlay/NetPlayBrowser.cpp

```diff
@@ -323,3 +328,51 @@ void NetPlayBrowser::accept()
 
   emit Join();
 }
+
+void NetPlayBrowser::SaveSettings() const
+{
+  auto& settings = Settings::Instance().GetQSettings();
+
+  settings.setValue(QStringLiteral("netplaybrowser/geometry"), saveGeometry());
+
+  settings.setValue(QStringLiteral("netplaybrowser/region"), m_region_combo->currentText());
+  settings.setValue(QStringLiteral("netplaybrowser/name"), m_edit_name->text());
+  settings.setValue(QStringLiteral("netplaybrowser/game_id"), m_edit_game_id->text());
+  QString visibility(QStringLiteral("all"));
+  if (m_radio_public->isChecked())
+    visibility = QStringLiteral("public");
+  else if (m_radio_private->isChecked())
+    visibility = QStringLiteral("private");
+  settings.setValue(QStringLiteral("netplaybrowser/visibility"), visibility);
+  settings.setValue(QStringLiteral("netplaybrowser/hide_incompatible"),
+                    m_check_hide_incompatible->isChecked());
+  settings.setValue(QStringLiteral("netplaybrowser/hide_ingame"), m_check_hide_ingame->isChecked());
+}
+
+void NetPlayBrowser::RestoreSettings()
+{
+  const auto& settings = Settings::Instance().GetQSettings();
+
+  const QByteArray geometry =
+      settings.value(QStringLiteral("netplaybrowser/geometry")).toByteArray();
+  if (!geometry.isEmpty())
+    restoreGeometry(geometry);
+
+  const QString region = settings.value(QStringLiteral("netplaybrowser/region")).toString();
+  if (m_region_combo->findText(region) != -1)
```

Comment:
Why not just compare the values?

Reply:
My intention was to only set the text if the text was a valid option of the combobox.

This is to avoid leaving the combobox in an invalid state, if the list of regions are ever renamed, but the user's settings still contain an old value.

I can possibly define a variable, with a relevant name, to make this more obvious:

```cpp
const bool valid_region = m_region_combo->findText(region) != -1;
if (valid_region)
  m_region_combo->setCurrentText(region);
```

Thoughts?

Reply:
Sure, that sounds good to me.